walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -117,4 +95,65 @@
     return timeRanges;
   }
 
+  /**
+   * Returns a list containing all the TimeRanges that the given collection of attendees are available for given
+   * a collection of their events for that day.
+   */
+  private List<TimeRange> determineAvailableTime(Collection<Event> events, Collection<String> attendees, long meetingDuration) {
+    Iterator<TimeRange> unavailableTimeIterator = determineUnavailableTime(events, attendees).iterator();
+    TimeRange currentEvent = (unavailableTimeIterator.hasNext()) ?  unavailableTimeIterator.next() : null;
+    if (currentEvent == null) {  // There are no events in the day.
+      return Arrays.asList(TimeRange.WHOLE_DAY);
+    } else {
```

Comment:
[Nit] Once you have an early return you typically remove the "else" condition as it is not longer needed.  In other words instead of

`if (x) {
  return stuff;
} else {
  doStuff()
}`

simply use

`if (x) {
  return stuff
}
doStuff()`

The idea is that you put all the preconditions at the beginning of a function and remove nesting.  There are a lot of discussions on this topic and how much it improves readability (or in some cases may even be a bad idea) but an good introduction can be found at https://www.itamarweiss.com/personal/2018/02/28/return-early-pattern.html

Reply:
I have found a few other places where I can implement this. Thanks!